src/gitops_connector.py

```diff
@@ -26,6 +26,11 @@ def __init__(self):
 
     def process_gitops_phase(self, phase_data, req_time):
         if self._gitops_operator.is_supported_message(phase_data):
+            repo_n = self._gitops_operator.get_repo_name(phase_data)
+            logging.debug(f'azdoPrRepoName: {repo_n}')
+            if repo_n != '':
+                self._gitops_operator.callback_url = self._git_repository.set_git_repository(repo_n)
+                self._cicd_orchestrator.set_git_repository(repo_n)
```

Comment:
The GitOps connector instance is shared across threads. It is very possible that another thread, initiated by an alert configured for the different application/repo, will override these self settings. So the message from thread 1 will go to the repo from thread 2.

Reply:
Besides, forwarding the alerts to AZDO, GitOps connector detects abandoned/rejected PRs to the GitOps repo and notifies the agentless waiting task to fail, so the whole promotional flow gets failed. This logic is implemented on top of having a single repo.

IMO, if we want to support multiple applications, we have the options:
 - Have an instance of GitOps Connector per application
 - Enhance the GitOps Connector to be configured with multiple repos (e.g. in config maps) and like you suggested, reference these repos in the alert definition